subsys/audio_modules/audio_module_template/audio_module_template.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright(c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include "audio_module_template.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include <errno.h>
+#include "audio_defines.h"
+#include "audio_module.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(audio_module_template, CONFIG_AUDIO_MODULE_TEMPLATE_LOG_LEVEL);
+
+static int audio_module_template_open(struct audio_module_handle_private *handle,
+				      struct audio_module_configuration const *const configuration)
+
+{
+	struct audio_module_handle *hdl = (struct audio_module_handle *)handle;
+	struct audio_module_template_context *ctx =
+		(struct audio_module_template_context *)hdl->context;
+	struct audio_module_template_configuration *config =
+		(struct audio_module_template_configuration *)configuration;
+
+	/* Perform any other functions required to open the module. */
+
+	/* For example: Clear the module's context.
+	 *              Save the initial configuration to the module context.
+	 */
+	memset(ctx, 0, sizeof(struct audio_module_template_context));
+	memcpy(&ctx->config, config, sizeof(struct audio_module_template_configuration));
+
+	LOG_DBG("Open %s module", hdl->name);
+
+	return 0;
+}
+
+static int audio_module_template_close(struct audio_module_handle_private *handle)
+{
+	struct audio_module_handle *hdl = (struct audio_module_handle *)handle;
+	struct audio_module_template_context *ctx =
+		(struct audio_module_template_context *)hdl->context;
+
+	/* Perform any other functions required to close the module. */
+
+	/* For example: Clear the context data */
+	memset(ctx, 0, sizeof(struct audio_module_template_context));
+
+	LOG_DBG("Close %s module", hdl->name);
+
+	return 0;
+}
+
+static int audio_module_template_configuration_set(
+	struct audio_module_handle_private *handle,
+	struct audio_module_configuration const *const configuration)
+{
+	struct audio_module_template_configuration *config =
+		(struct audio_module_template_configuration *)configuration;
```

Comment:
Here we also have a cast which may not be required. Ref. earlier comment.

Reply:
See earlier comment.